pytools/tag.py

```diff
@@ -195,34 +236,10 @@ class Taggable:
     """
 
     def __init__(self, tags: TagsType = frozenset()):
-        # For performance we assert rather than
-        # normalize the input.
         assert isinstance(tags, FrozenSet)
         assert all(isinstance(tag, Tag) for tag in tags)
+        check_tag_uniqueness(tags)
```

Comment:
@inducer I had to add these back in to get the test suite to pass. This is because you perform error handling in the unit tests: https://github.com/inducer/pytools/blob/main/test/test_pytools.py#L482-L496

Reply:
I'd say let's update the unit tests and document the policy on where input checking is to be done (i.e. not the constructor).

Reply:
@inducer I've gone ahead and updated the documentation and unit tests. I figured it's best to use `check_tag_uniqueness` for input-checking, and so I've documented that in the `Taggable` constructor.

I also modified the `check_tag_uniqueness` slightly to include an `isinstance(tag, Tag)` (since we already loop through them, might as well check they make sense). I removed the corresponding line from `normalize_tags` since it's always used internally and passed through `check_tag_uniqueness` in the relevant routines. Let me know if you think this is okay!